rewrite-java-test/src/test/java/org/openrewrite/java/format/WrappingAndBracesTest.java

```diff
@@ -956,33 +992,1829 @@ void annotationWrappingWithNulls() {
+          ),
+          java(
+            """
+              @SuppressWarnings(value = "all", justification = "test")
+              class Test {
+              }
+              """,
+            """
+              @SuppressWarnings(value = "all",
+                                justification = "test")
+              class Test {
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void annotationParametersOpenNewLine() {
+        rewriteRun(
+          autoFormat(
+            spaces -> spaces,
+            wrap -> wrap.withAnnotationParameters(wrap.getAnnotationParameters().withWrap(WrapAlways).withOpenNewLine(true))
+          ),
+          java(
+            """
+              @SuppressWarnings(value = "all", justification = "test")
+              class Test {
+              }
+              """,
+            """
+              @SuppressWarnings(
+                      value = "all",
+                      justification = "test")
+              class Test {
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void annotationParametersCloseNewLine() {
+        rewriteRun(
+          autoFormat(
+            spaces -> spaces,
+            wrap -> wrap.withAnnotationParameters(wrap.getAnnotationParameters().withWrap(WrapAlways).withCloseNewLine(true))
+          ),
+          java(
+            """
+              @SuppressWarnings(value = "all", justification = "test")
+              class Test {
+              }
+              """,
+            """
+              @SuppressWarnings(value = "all",
+                      justification = "test"
+              )
+              class Test {
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void annotationParametersAllOptions() {
+        rewriteRun(
+          autoFormat(
+            spaces -> spaces,
+            wrap -> wrap.withAnnotationParameters(wrap.getAnnotationParameters()
+              .withWrap(WrapAlways)
+              .withOpenNewLine(true)
+              .withCloseNewLine(true))
+          ),
+          java(
+            """
+              @SuppressWarnings(value = "all", justification = "test")
+              class Test {
+              }
+              """,
+            """
+              @SuppressWarnings(
+                      value = "all",
+                      justification = "test"
+              )
+              class Test {
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void annotationParametersDoNotWrap() {
+        rewriteRun(
+          autoFormat(
+            spaces -> spaces,
+            wrap -> wrap.withAnnotationParameters(wrap.getAnnotationParameters().withWrap(DoNotWrap))
+          ),
+          java(
+            """
+              @SuppressWarnings(value = "all", justification = "test")
+              class Test {
+              }
+              """
+          )
+        );
+    }
```

Comment:
Should we add a before text block here that's not conforming and then gets adjusted?